collect_app/src/main/java/org/odk/collect/android/widgets/BarcodeWidget.java

```diff
@@ -88,14 +89,20 @@ public IAnswerData getAnswer() {
     @Override
     public void setBinaryData(Object answer) {
         String response = (String) answer;
-        if (response != null) {      // It looks like the answer is not set to null even if no barcode captured, however it seems prudent to check
-            response = response.replaceAll("\\p{C}", "");
-        }
-        stringAnswer.setText(response);
+        stringAnswer.setText(stripInvalidCharacters(response));
 
         widgetValueChanged();
     }
 
+    // Remove control characters, invisible characters and unused code points.
+    @Contract("null -> null; !null -> !null")
```

Comment:
Ooh, nice!

Reply:
The IDE recommended that. I'm curious to see how it helps with code cleanliness - maybe it can eliminate some null pointer checks?